### ISSUE TITLE
poisson_conf_interval broken on older versions of scipy

### ISSUE THREAD

**Issue by embray** [astropy/astropy#5064]
The test for `poisson_conf_interval` added in #4622 fails for me because my scipy version 0.13.2, which is admittedly a bit old but not *that* old either, does not have `scipy.special.factorial`.

At the very least this function should have a version check for scipy with an appropriate error message, and the test skipped.  I looked in the docs and we don't actually specify a minimum scipy version, so for now we're stuck with it.

**Comment by astrofrog** [comment]
@hamogu - is there any reason you didn't use ``math.factorial`` instead? (speed?)

**Comment by embray** [comment]
`math.factorial` is probably useless

**Comment by astrofrog** [comment]
@embray - in what sense do you mean? Performance-wise, it's much better than ``scipy.special.factorial``:

```
In [18]: %timeit  np.float(math.factorial(33))
The slowest run took 9.38 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 934 ns per loop

In [19]: %timeit factorial(33)
The slowest run took 7.16 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 16 µs per loop
```

**Comment by hamogu** [comment]
in e.g. eqn 8, `n` is actually an array. Using `math.factorial` would require extra for loops for this case.

**Comment by pllim** [comment]
@embray , OMG... 0.13...

**Comment by hamogu** [comment]
We currently require numpy 1.6 or later, which would correspond to scipy 0.10 and with the 1.2 release we will up that to numpy 1.7 or later, which was released when scipy 0.11 was the most current version.
Thus, expecting astropy to work with Scipy 0.11 is not an unreasonable idea.
Dates taken from:
https://www.scipy.org/past-news.html

**Comment by embray** [comment]
@astrofrog
> in what sense do you mean? Performance-wise, it's much better than scipy.special.factorial
I'm sure it is for single integers....

**Comment by astrofrog** [comment]
I don't think it is 1.2-critical, but I think we should get a fix in for 1.2.1

**Comment by MSeifert04** [comment]
#5271 is merged, I think this can be closed.
